staging/src/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters/errors.go

```diff
@@ -67,6 +68,10 @@ func forbiddenMessage(attributes authorizer.Attributes) string {
 		return fmt.Sprintf("User %q cannot %s resource %q in API group %q in the namespace %q", username, attributes.GetVerb(), resource, attributes.GetAPIGroup(), ns)
 	}
 
+	if tenant := attributes.GetTenant(); len(tenant) > 0 {
+		return fmt.Sprintf("User %q cannot %s resource %q in API group %q in the tenant %q", username, attributes.GetVerb(), resource, attributes.GetAPIGroup(), tenant)
+	}
+
```

Comment:
This is a special rule enforced on the client side. Usually access rules should be enforced on server side.  I'm wondering if it's possible to handle this way:
1. The explicit "--tenant" parameter in command line alway takes precedence over context configuration. So if users specify "--tenant system" then we emit URL "/tenants/system/...", nothing special.
2. On server side we recognize "system" as a valid space name (probably we already did so?).